pkg/shares/non_interactive_defaults.go

```diff
@@ -83,10 +83,31 @@ func roundUpBy(cursor, v int) int {
 	}
 }
 
-// MinSquareSize returns the minimum square size that can contain shareCount
-// number of shares.
-func MinSquareSize(shareCount int) int {
-	return RoundUpPowerOfTwo(int(math.Ceil(math.Sqrt(float64(shareCount)))))
+// SubTreeWidth determines how many shares should be included when creating
```

Comment:
Also, don't we call this the `MaxSubTreeWidth` since a share commitment could be made of subtree roots of a smaller width (where width is the amount of leaf nodes in the subtree right?)

Reply:
> By this do you mean how many shares should be in a subtree root that gets used in the share commitment

yeah exactly, given the size of the blob in shares, how many leaves (width) should be in the subtree. 

>So if it's a leaf node the answer is 1 share and if it's an intermediary node it could be 2, 4, 8 or more shares that constitute the subtree root.

the input should be size of the blob determined in shares, tried to clarify in 643a18e2dd68a9e651e42ff359b961b0ce61d959

The width determined by this function is the width that must be used, correct? If that's the case, then I don't think max is appropriate.

Reply:
Not quite. Say there is a blob of 129 shares (and a threshold of 128). The result is 2 however the set of subroots that make the share commitment (AFAIU), is going to be [2, 1, 1, 1, 1, 1, ....] i.e. 1x2 and 127x1. Thus here 2 is the max and the rest is 1

Reply:
Actually it might be [2,2,2,2,2,2....,1] (but still 2 is the max width as there are subroots there of a smaller width